### PR TITLE
[PAY-779][PAY-781] Add premium track page and tile on mobile - Part 2

### PR DESCRIPTION
**Pull request by sddioulde**
### Description

Part 2 of the work that adds premium track page and tile to mobile.
https://github.com/AudiusProject/audius-client/pull/2818 will merge into this one and this one will merge into https://github.com/AudiusProject/audius-client/pull/2820.

### Dragons

n/a

### How Has This Been Tested?

local ios dapp against stage

### How will this change be monitored?

n/a

### Feature Flags ###

existing premium content feature flags

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-pc-mobile-track-page-tile-2

File: packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx [hunk 1]
```diff
@@ -63,31 +75,176 @@ const useStyles = makeStyles(({ palette, spacing, typography }) => ({
...
+  const styles = useStyles()
+
+  const { nftCollection, followee, tippedUser } =
+    useSpecialAccessEntity(premiumConditions)
+
+  const collectionLink = useMemo(() => {
+    if (!nftCollection) return ''
+
+    const { chain, address, externalLink } = nftCollection
+    if (chain === Chain.Eth && 'slug' in nftCollection!) {
```
Comment (dharit-tan) [inline packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx:156]: Hm why do we need the ! here if we already have the null check above?

File: packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx [hunk 2]
```diff
@@ -63,31 +75,176 @@ const useStyles = makeStyles(({ palette, spacing, typography }) => ({
...
+        <View style={styles.descriptionContainer}>
+          <Text>
+            <Text style={styles.description}>{messages.ownerTipGated}</Text>
+          </Text>
+        </View>
+      )
+    }
+
+    // should not reach here
+    return null
```
Comment (dharit-tan) [inline packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx:206]: Is there a better way to represent this? Maybe at least a `console.error()`?

File: packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx [hunk 3]
```diff
@@ -1,24 +1,109 @@
...
+  renderDescription,
+  isUnlocking
+}: NoAccessProps) => {
+  const styles = useStyles()
+  const neutral = useColor('neutral')
+
+  return (
+    <View style={styles.root}>
+      <View style={styles.titleContainer}>
+        <IconLock fill={neutral} width={16} height={16} />
```
Comment (dharit-tan) [inline packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx:99]: Just double-checking that design wants this to always be `neutral` color regardless of theme?

File: packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx [hunk 4]
```diff
@@ -30,12 +115,168 @@ export const DetailsTileNoAccess = ({
...
-    return <DetailsTileUnlockingSection premiumConditions={premiumConditions} />
-  }
+  const { nftCollection, followee, tippedUser } =
+    useSpecialAccessEntity(premiumConditions)
+
+  const collectionLink = useMemo(() => {
+    if (!nftCollection) return ''
+
+    const { chain, address, externalLink } = nftCollection
+    if (chain === Chain.Eth && 'slug' in nftCollection!) {
```
Comment (dharit-tan) [inline packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx:129]: Ditto above re: !

File: packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx [hunk 5]
```diff
@@ -30,12 +115,168 @@ export const DetailsTileNoAccess = ({
...
+      return `https://opensea.io/collection/${nftCollection.slug}`
```
Comment (dharit-tan) [inline packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx:130]: Also if logic is the same, should we pull it out to a separate function?

File: packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx [hunk 6]
```diff
@@ -30,12 +115,168 @@ export const DetailsTileNoAccess = ({
...
+          <Text>
+            <Text style={styles.description}>
+              {messages.lockedTipGatedPrefix}
+            </Text>
+            <Text style={styles.description}>{tippedUser.name}</Text>
+            <UserBadges
+              badgeSize={16}
+              user={tippedUser}
+              nameStyle={styles.description}
+              hideName
```
Comment (dharit-tan) [inline packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx:190]: Same comment as before about `nameStyle` being used alongside `hideName`.

**Review by dharit-tan — APPROVED** [review]

File: packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx (hunk 1 above)
Comment (sddioulde) [reply packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx:156]: good q. we shouldn't need

File: packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx (hunk 2 above)
Comment (sddioulde) [reply packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx:206]: yeah let me do a console.warn here thx

File: packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx (hunk 5 above)
Comment (sddioulde) [reply packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx:130]: fair

File: packages/common/src/store/premium-content/sagas.ts [hunk 7]
```diff
@@ -315,7 +315,7 @@ function* updateCollectibleGatedTrackAccess(
   }
 }
 
-const PREMIUM_TRACK_POLL_FREQUENCY = 5000
+const PREMIUM_TRACK_POLL_FREQUENCY = 1000
```
Comment (piazzatron) [inline packages/common/src/store/premium-content/sagas.ts:318]: good candidate for remote config imo

File: packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx [hunk 8]
```diff
@@ -63,31 +75,176 @@ const useStyles = makeStyles(({ palette, spacing, typography }) => ({
...
-}: {
-  premiumConditions: PremiumConditions
-}) => {
+type HasAccessProps = {
+  renderDescription: () => ReactNode
+  nftCollection?: Nullable<
+    PremiumConditionsEthNFTCollection | PremiumConditionsSolNFTCollection
+  >
+  followee?: Nullable<User>
+  tippedUser?: Nullable<User>
```
Comment (piazzatron) [inline packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx:90]: nit: a bit awks to have something that's both nullable & possible undefined, do we need both?

File: packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx [hunk 9]
```diff
@@ -63,31 +75,176 @@ const useStyles = makeStyles(({ palette, spacing, typography }) => ({
...
+
+  const { nftCollection, followee, tippedUser } =
+    useSpecialAccessEntity(premiumConditions)
+
+  const collectionLink = useMemo(() => {
+    if (!nftCollection) return ''
+
+    const { chain, address, externalLink } = nftCollection
+    if (chain === Chain.Eth && 'slug' in nftCollection!) {
+      return `https://opensea.io/collection/${nftCollection.slug}`
```
Comment (piazzatron) [inline packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx:157]: is there web code that could be shared here?

File: packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx (hunk 2 above)
Comment (piazzatron) [reply packages/mobile/src/components/details-tile/DetailsTileHasAccess.tsx:206]: could model this with a discriminated union and then the compiler would enforce that it's impossible for this to happen :)

File: packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx [hunk 10]
```diff
@@ -30,12 +115,168 @@ export const DetailsTileNoAccess = ({
...
+    return ''
+  }, [nftCollection])
+
+  const { onPress: handlePressCollection } = useLink(collectionLink)
+
+  const renderLockedDescription = useCallback(() => {
+    if (nftCollection) {
+      return (
+        <View style={styles.descriptionContainer}>
+          <Text style={styles.description}>
```
Comment (piazzatron) [inline packages/mobile/src/components/details-tile/DetailsTileNoAccess.tsx:145]: I didn't go through this w a super fine toothed comb but it seems like there's a fair bit of of commonality between the rendered components here in each branch (i.e. `.descriptionContainer` + nested `.description`) and also in the `HasAccess` version of this component - is there any way to consolidate this at all?

**Review by piazzatron — APPROVED** [review]

**Pull request merged**